sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosClientState.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.CosmosBridgeInternal;
+import com.azure.cosmos.implementation.caches.AsyncCache;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+public class CosmosClientState implements Serializable {
+    public byte[] collectionInfoByNameCache;
+    public byte[] collectionInfoByIdCache;
+
+    public CosmosClientState() {
+        System.out.println("initialized Cosmos Client State");
+    }
+
+    public void serialize(CosmosAsyncClient client) throws IOException {
+        RxDocumentClientImpl documentClient = (RxDocumentClientImpl) CosmosBridgeInternal.getAsyncDocumentClient(client);
```

Comment:
How will be "backward compatibility" mnaged?

Reply:
I am assuming you are referring to serializing caches of SDK from an older version of the SDK and trying to deserialize on newer version of the SDK. 

Assuming I understood the question correctly.
1) each of the component that we serialize/deserialize has a "serialVersionUID" which ensures the serialization code and deserialization code are on the same version. If they are not Java serialization will throw exception.

see this please:
_The serialization runtime associates with each serializable class a version number, called a serialVersionUID, which is used during deserialization to verify that the sender and receiver of a serialized object have loaded classes for that object that are compatible with respect to serialization. If the receiver has loaded a class for the object that has a different serialVersionUID than that of the corresponding sender's class, then deserialization will result in an InvalidClassException. A serializable class can declare its own serialVersionUID explicitly by declaring a field named "serialVersionUID" that must be static, final, and of type long:_
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html

2) if this happens in a spark cluster we have a bigger problem as it means different nodes of the spark cluster are on different versions of the spark-connector, in this situation many other things will go wrong too.